climada/hazard/base.py

```diff
@@ -1393,3 +1442,167 @@ def _read_att_excel(self, file_name, var_names):
         self.intensity = sparse.csr_matrix(dfr.values[:, 1:num_events + 1].transpose())
         self.fraction = sparse.csr_matrix(np.ones(self.intensity.shape,
                                                   dtype=np.float))
+
+    def concatenate(self, haz_src, append=False, centroids=None):
+        """
+        Concatenate events of several hazards of same type.
+
+        This is a wrapper method for hazard.base.concatenate_hazard
+        to ensure backwards compatibility.
+
+        Parameters
+        ----------
+        haz_src: list(climada.hazard.Hazard())
+            Hazard instances of the same hazard type
+        append : bool, optional
+            If True, append the haz_src to current object. If False,
+            replace all attributes from current objects by concatenated
+            hazards from haz_src. The default is False.
+        centroids: climada.hazard.Centroids(), optional
+            Centroids instance on which to map the concatenated hazard.
+            Default is None.
+
+        Returns
+        -------
+        None.
+
+        """
+        if append:
+            haz_src = [self] + haz_src
+        self.__dict__ = copy.deepcopy(
+            concatenate_hazard(haz_src, centroids).__dict__
+            )
+
+
+def join_centroids(haz_list):
+    """
+    Find centroids comon to all.
+
+    Centroids of all hazards must either all be rasters with the same
+    resolution, or all be points.
+
+    The centroids of all hazards are combined together. If points,
+    all points are combined. If rasters, a global raster is defined.
+
+    Parameters
+    ----------
+    haz_list : list(climada.hazard.Hazard())
+        List of hazard with centroids to join
+
+    Returns
+    -------
+    centroids : climada.hazard.Centroids()
+        Centroids containing the centroids of all hazards.
+
+    """
+    centroids = copy.deepcopy(haz_list[0].centroids)
+    for haz in haz_list[1:]:
+        if not centroids.equal(haz.centroids):
+            centroids.append(haz.centroids)
+    centroids.remove_duplicate_points()
+    return centroids
+
+def concatenate_hazard(haz_list, centroids=None):
```

Comment:
I'd really like to add the parameter `threshold`, which is then passed to `u_coord.assign_coordinates`.
- Right now it's unsettable, which means `assign_coordinates` uses its default value of 100 when matching centroids. The user should at least know this.
- If threshold is settable, then setting it to zero gives you a neat way to combine hazard objects while only retaining data defined exactly at the centroids you care about. (Though you could just subset the hazards before you combine them)

Additionally, I think the current code will fail when it can't find a centroid within the default `threshold` distance, since it'll try to map things to index -1.

Reply:
Again, good points! I can easily add that possibility. 

Note however that the use case you propose would not work (precisely for the reason you said "Additionally, I think the current code will fail when it can't find a centroid within the default threshold distance, since it'll try to map things to index -1.").

Hence, a use case would be  actually the opposite: if you want to assign hazard centroids that are very fare from
the user defined centroids, you can make threshold larger. I  do not really see that this would lead to good results, but it is good for the user to be able to tune that. 

I could catch the case when centroids are beyond the threshold and inform the user that they cannot merge.

Reply:
"If threshold is settable, then setting it to zero gives you a neat way to combine hazard objects while only retaining data defined exactly at the centroids you care about. (Though you could just subset the hazards before you combine them)"
This is not possible with the current implementation and would represent some extra work that for me doesn't fit into this pull request as it would require the possibility to select subsets of hazards.

Reply:
I forgot to suggest that when the method can't find a nearby centroid within the threshold (and returns -1) I think you should drop that centroid (with a logging message). It wouldn't be too hard to snip out those columns when you're rearranging the sparse matrix.

You're right though - if you throw an error when you get a -1, then that second bullet point doesn't work. I'm also fine with you throwing an error - it's not a bad idea to make it the user's responsibility that their inputs will map sensibly.

Reply:
I think I prefer the error message. Otherwise, by simply dropping centroids the data is modified in a rather intransparent way.

Reply:
The threshold has been added and a warning if beyond the threshold too. f3a872c2da5a8af9d270ba1d7b994d378d3607a9
Also for `hazard.concatenate` here 8e5422a2b8882f01874200312a7cadc0831f25e3

Reply:
Nice